core/store/tx_manager_test.go

```diff
@@ -40,7 +41,11 @@ func TestTxManager_CreateTx_Success(t *testing.T) {
 	})
 	assert.NoError(t, app.StartAndConnect())
 
+	gomega.NewGomegaWithT(t).Eventually(func() bool {
+		return manager.Connected()
+	}).Should(gomega.Equal(true))
 	require.True(t, manager.Connected())
```

Comment:
Can't you remove this line if you eventually assert it's true in the above line?

Reply:
Interesting to find this to be an issue, I'm a bit surprised but it makes sense. Should we drop a comment suggesting a future more global fix, since this pattern exists in other tests.

Reply:
Yeah I was keeping that in to ensure the test aborts, did some testing and gomega does hard fail the test. So removing now.